### PR TITLE
Refactor processor script

### LINKED ISSUE #11: Refactor processor script
@cmccandless I was just wondering, if you have time, would you like to refactor the processor script in this repo with the changes you made here: https://github.com/exercism/c-test-runner/pull/8? If not, would it be okay if I made the changes?

### PR DESCRIPTION
**Pull request by cmccandless**
Resolves #11
Duplication of efforts in https://github.com/exercism/c-test-runner/pull/8

For my first pass here, I've simply copy-pasted the finished version from the c-test-runner. If this track has different requirements that I failed to notice in the current version, I can revert this commit and perform the refactoring process manually again to retain unique functionality.

### CODE REVIEW

File: process_results.py [hunk 1]
```diff
@@ -2,70 +2,63 @@
...
-                    done = True
-                    break
-                case = {}
-                i += 1
-            else:
-                buf += line
-    if not done:
+    pattern = r"(?m)^((?P<file>.*test_.*\.c):\d+:(?P<name>\w+):(?P<status>PASS|FAIL)(?:: (?P<message>.*))?)$"
+    text = filepath.read_text()
+    text = text[20:]
```
Comment (bergjohan) [inline process_results.py:30]: I think the only difference is that this line needs to be removed. The C makefile outputs a line of text, which is not the case for the x86-64 assembly makefile.

**Comment by bergjohan** [comment]
Hmm, seems to be failing on the next test. One more difference is that the x86-64 test files are named `<slug>_test.c` and not `test_<slug>.c`. If that's not the problem I'll have a look later tonight, when I'm at my computer.

**Comment by bergjohan** [comment]
Thank you! :)

Did the tests pass? I can't see any test run..

**Comment by cmccandless** [comment]
They didn't run; I don't know why

**Comment by cmccandless** [comment]
I think GitHub may be experiencing issues; I pushed a new empty commit to my branch to trigger CI builds, and the commit shows [on my repo](https://github.com/cmccandless/x86-64-assembly-test-runner/commits/patch-1), but not here.

**Comment by cmccandless** [comment]
Confirmed GitHub issue: https://www.githubstatus.com/incidents/phnch1rww464

**Comment by cmccandless** [comment]
@bergjohan Checks finally ran and passed

**Comment by bergjohan** [comment]
Great work! :) Thanks again!

**Comment by cmccandless** [comment]
This is ready for merge if there are no further comments

**Comment by bergjohan** [comment]
Merged!

**Pull request merged**